saleor/core/utils/random_data.py

```diff
@@ -970,9 +975,13 @@ def create_orders(how_many=10):
 def create_product_promotions(how_many=5):
     for _ in range(how_many):
         promotion = create_fake_promotion()
-        update_products_discounted_prices_of_promotion_task.delay(promotion.pk)
         yield f"Promotion: {promotion}"
 
+    # recalculation is handled by celery beat, so we trigger it manually, to receive the
+    # correct amounts in random data created by saleor.
+    update_variant_relations_for_active_promotion_rules_task()
```

Comment:
It was done fully async previously. Don't think that we should worry about it. The batch of promotion rules is 100. The batch of products is 2k. So it will be enough in most cases. If you're worried about this, we can consider a change on main.